src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -386,15 +386,20 @@ public async Task SaveChangesAsync(IList<UnspentOutputs> unspentOutputs, IEnumer
 
                         FetchCoinsResponse result = await this.inner.FetchCoinsAsync(new[] { unspent.TransactionId }).ConfigureAwait(false);
 
+                        foreach (var utxo in result.UnspentOutputs)
+                        {
+                            this.logger.LogTrace("UTXO fetched from disk: {0}", utxo);
+                        }
+
                         UnspentOutputs unspentOutput = result.UnspentOutputs[0];
 
                         cacheItem = new CacheItem();
                         cacheItem.ExistInInner = unspentOutput != null;
                         cacheItem.IsDirty = false;
-
                         cacheItem.UnspentOutputs = unspentOutput?.Clone();
 
                         this.cachedUtxoItems.TryAdd(unspent.TransactionId, cacheItem);
+                        this.logger.LogTrace("CacheItem added to cache {0}:{1}", nameof(cacheItem.ExistInInner), cacheItem.ExistInInner);
```

Comment:
This  does not help us much we need to know the identifier.

Basically we need to know every UTXO (not trx) that is inserted and removed form the cache.

Reply:
OK will fix :)